UI/js/controllers.js

```diff
@@ -52,45 +52,64 @@ dashboardApp.factory('Courses', function ($http) {
   };
 });
 
-dashboardApp.controller('coursesController', ['Courses', '$rootScope', '$scope', function (Courses, $rootScope, $scope) {
 
-  $scope.courses = [];
-  $scope.loading = true;
-
-  var url = 'courses/' + $rootScope.user + '.json';
-
-  Courses.getCourses(url).then(function (data) {
-    if (data.failure) {
-      $scope.courses.errors = data;
-      $scope.loading = false;
-    } else {
-      $scope.courses = data;
-      $scope.loading = false;
+dashboardApp.factory('Terms', function ($http) {
+  return {
+    getTerms: function (url) {
+      return $http.get(url, {cache: true}).then(
+        function success(result) {
+         if(result.data.Meta.httpStatus !==200){
+            result.errors = errorHandler(url, result);
```

Comment:
I see what you mean. There will be two types of errors. This one here will be caught by the servlet and reported in the result.data.Meta.httpStatus value -  but the transaction itself will be a success.  The function error(result) below catches the rest (that are truly reported as errors).